syntax = "proto3";

package syft.core.node.common.action;

import "proto/core/common/common_object.proto";
import "proto/core/pointer/pointer.proto";
import "proto/core/io/address.proto";

message RunFunctionOrConstructorAction {
  string path = 1;
  repeated syft.core.pointer.Pointer args = 2;
  map<string, syft.core.pointer.Pointer> kwargs = 3;
  syft.core.common.UID id_at_location = 4;
  syft.core.io.Address address = 5;
  syft.core.common.UID msg_id = 6;
}